src/components/DatePicker/Calendar.tsx

```diff
@@ -0,0 +1,297 @@
+import React, { useState, useEffect, useRef } from "react";
+import {
+  SvgIcChevronDown,
+  SvgIcChevronLeft,
+  SvgIcChevronRight,
+  SvgIcChevronUp,
+} from "../../assets/svg-components";
+import "./DatePicker.scss";
+import { months, years } from "../../utils/dateHandler";
+export interface CalendarProps {
+  dateVal?: string;
+  onDateClick: Function;
+  isRange?: boolean;
+  from: string;
+  to: string;
+  calendar: any;
+  selectedMonth: string;
+  selectedYear: string;
+  monthHandler: Function;
+  yearHandler: Function;
+  handleNextButtonClicked?: Function;
+  handlePreviousButtonClicked?: Function;
+  calendarId: string;
+}
+
+const Calendar = (props: CalendarProps) => {
+  const {
+    dateVal,
+    onDateClick,
+    isRange,
+    from,
+    to,
+    calendar,
+    selectedMonth,
+    selectedYear,
+    monthHandler,
+    yearHandler,
+    handleNextButtonClicked,
+    handlePreviousButtonClicked,
+    calendarId,
+  } = props;
+
+  const [days, setDays] = useState([
+    { name: "S", enum: 7 },
+    { name: "M", enum: 1 },
+    { name: "T", enum: 2 },
+    { name: "W", enum: 3 },
+    { name: "T", enum: 4 },
+    { name: "F", enum: 5 },
+    { name: "S", enum: 6 },
+  ]);
+  useEffect(() => {}, [props]);
```

Comment:
It seems that there is no callback function being executed inside the useEffect hook. If this is the case, consider removing the hook altogether or adding some code inside the function body. Otherwise, the hook may not be doing anything useful and can be removed.

Reply:
Fixed.